core/server/src/main/java/alluxio/worker/file/FileWorkerMasterSyncExecutor.java

```diff
@@ -147,11 +137,19 @@ public FilePersister(FileDataManager fileDataManager, long fileId, List<Long> bl
 
     @Override
     public void run() {
-      try {
+      if (mFileDataManager.needPersistence(mFileId)) {
+        // lock all the blocks of the file to prevent eviction
```

Comment:
Now that we're doing this asynchronously, the file could be evicted before the executor schedules this thread. Do we need to pin the file on the master side before sending the persist command? Do we guarantee that an async persist file won't be evicted before being persisted?